rmc/utils/generic.py

```diff
@@ -237,6 +237,166 @@ def process_context_ht(
+
+    :param overwrite_temp: Whether to overwrite temporary data.
+        If False, will read existing temp data rather than overwriting.
+        If True, will overwrite temp data.
+    :param keep_transcripts: Desired set of transcripts to keep from the context HT.
+        If set, function will filter to keep these trancripts only.
+        Default is None.
+    :param n_partitions: Desired number of partitions for context HT after filtering.
+        Default is 10,000.
+    :return: VEP context HT filtered to keep only transcript ID, protein number, and amino acid information.
+    """
+    logger.info(
+        "Reading in VEP context HT and filtering to coding effects in non-outlier"
+        " canonical transcripts..."
+    )
+    # Drop globals and select only VEP transcript consequences field
+    ht = process_context_ht(
+        filter_outlier_transcripts=True, add_annotations=False
+    ).select_globals()
+    ht = ht.select("transcript_consequences")
+    ht = ht.filter(
+        ~hl.literal(CSQ_NON_CODING).contains(
+            ht.transcript_consequences.most_severe_consequence
+        )
+    )
+    # Unnest annotations from context HT
+    ht = ht.transmute(
+        transcript=ht.transcript_consequences.transcript_id,
+        aa_start_num=ht.transcript_consequences.protein_start,
+        aa_end_num=ht.transcript_consequences.protein_end,
+        amino_acids=ht.transcript_consequences.amino_acids,
+    )
+
+    if keep_transcripts:
+        logger.info("Filtering to desired transcripts only...")
+        ht = ht.filter(hl.literal(keep_transcripts).contains(ht.transcript))
+
+    ht = ht.naive_coalesce(n_partitions)
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc/vep_amino_acids.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    return ht
+
+
+def get_ref_aa(
+    ht: hl.Table,
+    overwrite_temp: bool,
+    extra_aa_map: Dict[str, str] = {"*": "Ter", "U": "Sec"},
+    aa_to_remove: Set[str] = {"X"},
+) -> hl.Table:
+    """
+    Filter input HT to keep only reference amino acid information.
+
+    :param ht: Input Table. Should be HT output from `get_aa_from_context`.
+    :param overwrite_temp: Whether to overwrite temporary data.
+        If False, will read existing temp data rather than overwriting.
+        If True, will overwrite temp data.
+    :param extra_aa_map: Dictionary mapping any amino acid one letter codes to three letter codes.
+        Designed to capture any amino acids not present in `ACID_NAMES_PATH`.
+        Default is {"*": "Ter", "U": "Sec"}.
+    :param aa_to_remove: Any amino acid one letter codes to remove. Default is {"X"}.
+    :return: HT with one reference amino acid annotated per locus/transcript.
+    """
+    logger.info("Mapping amino acid one letter codes to three letter codes...")
+    aa_map = get_aa_map()
+    # Add any extra mappings into amino acid mapping dict
+    if extra_aa_map:
```

Comment:
that's true, but this seemed more flexible than either having an additional if check to avoid running this update if `extra_aa_map` is None / raising an error if `extra_aa_map` is None. what do you think?

Reply:
makes sense. No harm in keeping this logic. Just wondering if this logic was deliberate which it is, so all good :)